### ISSUE TITLE
[BUG] Issue with typescript typings of Model.scan

### ISSUE THREAD

**Issue by avikpal** [dynamoose/dynamoose#934]
I was following the documentation to run a scan along with a condition. It's resulting into a typescript compilation error

```
Property 'exec' does not exist on type 'Condition'.
```

this is the code snippet I'm using
```
public async fetchByUserName(username: string) {
        return await this.model.scan('username').contains(username).exec();
    }
```
tsconfig.json
```
{
    "compilerOptions": {
      "module": "commonjs",
      "esModuleInterop": true,
      "target": "es6",
      "moduleResolution": "node",
      "sourceMap": true,
      "outDir": "./dist",
      "rootDir" : "./src",
      "strict": true
    },
    "lib": ["es2015"],
    "include": ["src/*/*.ts"]
  }
```

### Environment:

Operating System: macOS Catalina
Operating System Version: 10.15.4
Node.js version (`node -v`): 10.16.0
NPM version: (`npm -v`): 6.9.0
Dynamoose version: 2.2.1

**Comment by avikpal** [comment]
Following is a stack trace for the same
```
at createTSError (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:434:12)
    at reportTSError (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:438:19)
    at getOutput (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:578:36)
    at Object.compile (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:775:32)
    at Module.m._compile (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:858:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/Users/avpal/Documents/personal/goldcrest/src/contollers/menu-controller.ts:17:27)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Module.m._compile (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:858:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/avpal/Documents/personal/goldcrest/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
```

**Comment by hardyscc** [comment]
@avikpal This PR #925 should fix your problem, Do u mind to test and review it? 
You can install that PR branch using:
```
npm install hardyscc/dynamoose --save
```

**Comment by pupca** [comment]
> @avikpal This PR #925 should fix your problem, Do u mind to test and review it?
> You can install that PR branch using:
> 
> ```
> npm install hardyscc/dynamoose --save
> ```

I can confirm, I am running into same issue as OP.

I am getting `Cannot find module 'dynamoose'.` when following your advise.

**Comment by hardyscc** [comment]
@pupca I just tested it works ok, can u try again?

Please make sure after `npm install` your dependencies look like this.
```
  "dependencies": {
    "dynamoose": "github:hardyscc/dynamoose",
  }
```

**Comment by polem** [comment]
> @pupca I just tested it works ok, can u try again?
> 
> Please make sure after `npm install` your dependencies look like this.
> 
> ```
>   "dependencies": {
>     "dynamoose": "github:hardyscc/dynamoose",
>   }
> ```

The `dist` folder is missing, with this method

**Comment by hardyscc** [comment]
@polem
The `prepare` scripts should run during npm install, 
It may due to some bug on previous version of npm, please try to install the latest one.

another way to test it is just 
```
git clone https://github.com/hardyscc/dynamoose.git
npm install
npm run build
```
then copy the ./dist folder over to ./node_modules/dynamoose/.

**Comment by fishcharlie** [comment]
@hardyscc Should this be closed from #983?

**Comment by hardyscc** [comment]
@fishcharlie yes please close it thanks

**Comment by lmanerich** [comment]
Any ETA to a new release?

**Comment by KyleMit** [comment]
@hardyscc  - can you clarify why that fixes this issue?  Is the suggestion to use your dynamoose fork for long term usage, or just diagnostics?


It looks like it just swaps out the library for yours.  I was thinking maybe it would add types or something?

![dynamoose in package.json](https://user-images.githubusercontent.com/4307307/100645191-51ccd500-330a-11eb-824e-9c8fa4c5c3f2.png)

Currently, the only [delta I see between your fork and the main repo](https://github.com/dynamoose/dynamoose/compare/master...hardyscc:master) is an update to the Model.md

**Comment by fishcharlie** [comment]
@KyleMit Correct. This was just released as part of the v2.4.0 release about a week ago.

The fix from @hardyscc above was temporary until I merged the PR and released a new version (both of which have been done).

Let me know if there are any additional problems.

**Comment by KyleMit** [comment]
Thanks @fishcharlie  & @hardyscc !!  Appreciate your work and time on this project :)
